stickyuploads/storage.py

```diff
@@ -12,4 +15,18 @@ def __init__(self):
 
     def url(self, name):
         """Files are not accessable via url."""
-        raise NotImplementedError()
\ No newline at end of file
+        raise NotImplementedError()
+
+    def get_available_name(self, name, max_length=None):
+        """Return path to name placed in two random directories"""
+        name = os.path.join(
```

Comment:
The reason why I want to put the file in nested directories is to prevent too many nodes in one directory and still enough random characters (in this case 6) for directory name to prevent file name collision.

With first directory only two letters long we will have max 676 directories in the temp directory. All possible combinations are 308 915 776 which is enough to prevent file names collision if the same file or files with the same name are uploaded multiple times.

Max number of sub directories inside a directory is 456 976 which is kind of a lot but IMHO this limit will hardly be reached.

If you think that this is too much the I can reduce the length of the sub directory but this will increase the chance of guessing path and steeling an uploaded file.